scripts/lib-franklin.js

```diff
@@ -775,6 +775,30 @@ export function decorateRenderHints(block) {
 export function isInternalPage() {
   return getHref().indexOf('/sidekick/blocks/') > 0 || getHref().indexOf('/_tools/') > 0;
 }
+
+function sendResizeEvent() {
+  const heightPx = window.innerHeight || document.documentElement.clientHeight;
+  const widthPx = window.innerWidth || document.documentElement.clientWidth;
+
+  const baseFontSize = parseFloat(getComputedStyle(document.documentElement).fontSize);
+  const heightRem = heightPx / baseFontSize;
+  const widthRem = widthPx / baseFontSize;
+
+  // Create a new custom event with custom parameters
+  const resizeEvent = new CustomEvent('viewportResize', {
+    detail: {
```

Comment:
Let me add deviceType, is it fine to continue having width & height as well so that consumers can have more fine grained control with in Desktop/Tablet/Mobile if needed.
I can see in the project, we have media queries with other breakpoints as well apart from 77rem, 62rem.

Reply:
Yes, makes sense, we can have both.

Reply:
@shrotia in my opinion this approach is just a beautified version of window.resize, it still introduces the problem of frequent event firing and layout thrashing.
you can definitely specify viewport sizes in an object in utils for better readability and it can be used in other places as well